Marlin/src/module/motion.cpp

```diff
@@ -573,26 +573,35 @@ void clean_up_after_endstop_or_probe_move() {
 
     #else
 
-      #if !HAS_SOFTWARE_ENDSTOPS || ENABLED(MIN_SOFTWARE_ENDSTOP_X)
+      #if HAS_SOFTWARE_ENDSTOPS || ENABLED(MIN_SOFTWARE_ENDSTOP_X)
```

Comment:
This is breaking the intended logic!

Reply:
So its intended to clamp to a soft limit even when soft limits are disabled?

Reply:
When they don't exist, the array is filled in with defaults.

Reply:
Fair enough, like I said when I passed through this briefly last night it just stuck out and I hadnt looked at it much yet. I would have expected no restriction if disabled. You see my reasoning for posting this before I had a chance to dig more above.

Reply:
A hint that it was intended: Any of the `ENABLED(Mxx_SOFTWARE_ENDSTOP_x)` will result in `HAS_SOFTWARE_ENDSTOPS` being true.